src/components/presentational/Grid/Grid.css

```diff
@@ -1,13 +1,27 @@
 .mdhui-grid {
     display: flex;
-    gap: 16px;
+    gap: var(--mdhui-padding-md);
+    flex-wrap: wrap;
 }
 
 .mdhui-grid.mdhui-grid-default-margin {
-    margin: 16px;
+    margin: var(--mdhui-padding-md);
 }
 
 .mdhui-grid .mdhui-grid-column {
     flex-grow: 0;
     flex-shrink: 0;
+}
+
+.mdhui-grid .mdhui-grid-column:empty {
+    display: none;
+}
+
+/* If there is only an empty card and no other non-empty elements in the column, hide the column */
+.mdhui-grid .mdhui-grid-column:has(> .mdhui-card:empty):not(:has(> *:not(:empty))) {
```

Comment:
Annoyingly, this CSS rule appears to not work on iOS safari due to some sort of bug.  Funny enough, if in developer tools I click edit on the CSS block for this style, but don't edit this rule at all, and just save, it works fine.  So something to do with the initial render of the webview or something